### ISSUE TITLE
Choice of Cluster values

### ISSUE THREAD

**Issue by laiadc** [konar1987/3D-QNet#1]
In the Transit function, you define an array called 'cluster', for different lenghts (L=4, 6, 7, 8). It seems to me that the choice of this cluster array highly influences the output and performance of the algorithm. I understand it is a hyperparameter of the algorithm. Do you have any idea of what is the best way to choose this array, based on the input image? Could you describe what the role of this cluster variable?

Thank you so much!

Best,

Laia.

**Comment by konar1987** [comment]
Dear Laia,

Thanks for your interest in our 3D-QNet. 

The clusters are adaptive and based on the optimization on input images. The cluster decises the boundaries for multi-level sigmoid activation function which we have proposed. 

If you have any further questions, please let me know in my mail id: ***@***.***


Best regards,

Debanjan

**Comment by darkEnvious** [comment]
Hi,

Firstly, thanks for making your codebase public - it is very helpful to browse through the code while reading your paper on 3-D Quantum-Inspired Self-Supervised Tensor Network for Volumetric Segmentation of Medical Images. 

I had a question on the above lines - what are the parameters in the code that get updated according to the learning process? Or is the provided code applying pre-defined Quantum-Inspired transformations and then using classical segmentation models? I was just curious and wanted to have a look at how this model achieves self-supervised learning.

Thanks in advance, and I am looking forward to hearing from you!

Regards,
Nirmal

**Comment by konar1987** [comment]
Dear Nirmal,

Thanks for your interest in our work.

The weight parameters are updated in a self-supervised fashion without any
training. We have proposed an intelligent 3D decomposition of weights
(Tucker Decomposition) method in the self-learning process.

Please let me know if you have any further questions.

Best regards,


Debanjan

**Comment by darkEnvious** [comment]
Hi, thanks a lot for your response!

I understand that the weight parameters are updated through an iterative process, as described in the Transit.m file. However, is the Tucker Decomposition method included in this code, or is it only proposed in the paper? I am curious to look at the differences between the non-tensor 3D-QNet and the tensor-powered 3D-QNet first-hand, and I would appreciate your insights here!

Thanks in advance!

Regards,
Nirmal

**Comment by konar1987** [comment]
Hi Nirmal, 

I am sorry, but I have lost your mail in my inbox, so there is a delay in getting back to you directly from Github. 
It is a Tucker-like decomposition as the current, previous, and next slice of 3D images suitable for semantic segmentation.  

If you have any further questions, please let me know.
